### DISCUSSION TITLE
How to debug custom extensions / where is console.log

### DISCUSSION THREAD

**Discussion by bittsitt** [foxglove/community#887] [Q&A] [answered]
I am writing a custom extension, and although I'm pretty sure I've missed something obvious, I have not been able to figure out where messages from console.log (or console.debug) end up.

I did find the "Enable panels and features for debugging Foxglove", and I now see Logs in the right sidebar where I can e.g. change level, but still not sure where it actually ends up?

If there are other tips to debugging extensions, feel free to share as well.

**Comment by achim-k** [comment, marked as answer]
You can open the developer tools with <kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>i</kbd>, there you will see messages logged with `console.log`

**Reply by bittsitt** [reply]
Ah, perfect. Thanks!
